litgpt/api.py

```diff
@@ -0,0 +1,219 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+#
+# This file implements the LitGPT Python API
+from pathlib import Path
+from typing import Any, List, Literal, Optional, Tuple, Union
+
+import torch
+import lightning as L
+from lightning.fabric.utilities.device_parser import _normalize_parse_gpu_input_to_list
+
+from litgpt.model import GPT  # needs to be imported before config
+from litgpt.config import Config
+from litgpt.tokenizer import Tokenizer
+from litgpt.generate.base import generate as generate_fn
+from litgpt.prompts import load_prompt_style, has_prompt_style, PromptStyle
+from litgpt.utils import (
+    extend_checkpoint_dir,
+    get_default_supported_precision,
+    load_checkpoint,
+)
+
+
+class LLM:
+    def __init__(
+        self,
+        model: GPT,
+        tokenizer: Tokenizer,
+        prompt_style: PromptStyle,
+        devices: Union[int, List[int]] = 1,
+        checkpoint_dir: Path = None,
+        fabric: L.Fabric = None
+    ) -> None:
+        self.model = model
+        self.preprocessor = Preprocessor(tokenizer, device=fabric.device)
+        self.devices = devices
+        self.prompt_style = prompt_style
+        self.checkpoint_dir = checkpoint_dir
+        self.fabric = fabric
+    """
+    LLM model class for inference, pretraining, and finetuning.
+
+    Example:
+        from litgpt.api import LLM
+
+        llm = LLM.load("microsoft/phi-2", device_type="cuda", devices=1)
+        text = llm.generate("What do Llamas eat?", top_k=1)
+        print(text)
+    """
+    @classmethod
+    def load(
+        cls,
+        model: str,
+        device_type: Literal["cpu", "cuda", "auto"] = "auto",
```

Comment:
I don't think something like `"cuda:0"` is supported in Fabric: https://lightning.ai/docs/fabric/stable/fundamentals/accelerators.html

```
# CPU (slow)
fabric = Fabric(accelerator="cpu")

# GPU
fabric = Fabric(accelerator="gpu", devices=1)

# GPU (multiple)
fabric = Fabric(accelerator="gpu", devices=8)

# GPU: Apple M1/M2 only
fabric = Fabric(accelerator="mps")

# GPU: NVIDIA CUDA only
fabric = Fabric(accelerator="cuda", devices=8)

# TPU
fabric = Fabric(accelerator="tpu", devices=8)
```

I think it's easier to aim for consistency here rather than reinventing the wheel and doing our own parsing. 
But thanks for raising that, I will change `device_type` to `accelerator`.